pkg/service/service.go

```diff
@@ -187,13 +190,76 @@ func (s *Service) GetCurrentTs() time.Time {
 	return time.Now()
 }
 
+func (s *Service) populateBackendMapV1FromV2(ipv4, ipv6 bool) error {
+	removeMap := func(m string) {
+		p := path.Join(bpf.MapPrefixPath(), m)
+		if _, err := os.Stat(p); !os.IsNotExist(err) {
+			if e := os.RemoveAll(p); e != nil {
+				log.WithError(err).WithField(logfields.Path, p).Warn("Error deleting map")
+			} else {
+				log.WithField(logfields.Path, p).Info("Deleted bpf map")
+			}
+		}
+	}
+	if ipv4 {
+		copyBackend4Entries := func(key bpf.MapKey, value bpf.MapValue) {
+			count := 0
+			backendKey := key.(*lbmap.Backend4KeyV2)
+			// Count entries with an ID greater than uint16 max, these entries will not be copied.
+			if backendKey.ID > math.MaxUint16 {
+				count = count + 1
+			} else {
+				backend4MapKey := lbmap.NewBackend4Key(backendKey.GetID())
+				lbmap.Backend4Map.Update(backend4MapKey, value.DeepCopyMapValue())
+			}
+			if count > 0 {
```

Comment:
This if-statement should be after calling `lbmap.Backend4MapV2.DumpWithCallback()`. Also, `count` should be outside the callback function. Otherwise, for each `ID > max(uint16)` you are going to log the warning msg.

Reply:
Good point, moved it outside of the callback func.